server/controllers/finance/fiscal.js

```diff
@@ -1,21 +1,76 @@
-var db      = require('./../../lib/db');
-var journal = require('./journal');
-var _       = require('lodash/lang');
+var db = require('../../lib/db');
+const NotFound = require('../../lib/errors/NotFound');
 
-exports.createFiscalYear = createFiscalYear;
-exports.fiscalYearResultat = fiscalYearResultat;
-exports.getFiscalYears = getFiscalYears;
 
-// GET /fiscal
-function getFiscalYears(req, res, next) {
+function lookupFiscalYear(id) {
   'use strict';
 
+  var sql =
+    `SELECT id, enterprise_id, number_of_months, label, start_date, 
+     previous_fiscal_year_id, locked, note
+     FROM fiscal_year 
+     WHERE id = ? `;
+
+  return db.exec(sql, id)
+    .then(function (rows) {
+      // Record Not Found !
+      if (rows.length === 0) {
+        throw new NotFound(`Record Not Found with id: ${id}`);
+      }
+
+      return rows[0];
+    });
+}
+
+// GET /fiscal
+function list (req, res, next) {
+  'use strict';
   var sql;
 
+  if (req.query.detailed === '1') {
+    sql =
+      `SELECT f.id, f.enterprise_id, f.number_of_months, f.label, f.start_date, 
+       f.previous_fiscal_year_id, f.locked, f.created_at, f.updated_at, f.note, DATE_ADD(start_date, INTERVAL number_of_months MONTH) AS end_date, 
+       f.user_id, u.first, u.last
+       FROM fiscal_year AS f
+       JOIN user AS u ON u.id = f.user_id
+       WHERE f.enterprise_id = ? `;
+
+    if(req.query.by && req.query.order){
+      sql += " ORDER BY " + req.query.by + " " + req.query.order + " ;";
+    }
+
+    db.exec(sql, [req.session.enterprise.id])
+    .then(function (rows) {
+      res.status(200).json(rows);
+    })
+    .catch(next)
+    .done();
+
+  } else {
```

Comment:
Instead of having two possible `db.exec()` calls, it would be better to have two possible SQL statements.  This makes the code easier to read (fewer branches in the logic).

Why  don't you try something like this?  (Pseudo-code)

```js
let sql;
let params = [];

// make a complex query
if (req.query.detailed === '1') {
  params = params.concat(req.session.enterprise.id);
  sql = `... f.enterprise_id = ? `;
 
  if (req.query.by && req.query.order) {
    params = params.concat(req.query.by, req.query.order);
    sql += ' ORDER BY ? ??;';
  }

// simple query
}  else {
  sql = 'SELECT id, label FROM fiscal_year;';
}

// execute the query
db.exec(sql, params)
.then(rows => {
  res.status(200).json(rows);
})
.catch(next)
.done(); 
```

Also, your JSHint should throw errors when you use `"`.  Make sure your strings use single-quotes (`'`).